Code/Framework/AzCore/AzCore/Math/MathUtils.h

```diff
@@ -461,6 +462,42 @@ namespace AZ
         return (azisfinite(x) != 0);
     }
 
+    //! Returns the value divided by alignment, where the result is rounded up if the remainder is non-zero.
+    //! Example: alignment: 4
+    //! Value:  0 1 2 3 4 5 6 7 8
+    //! Result: 0 1 1 1 1 2 2 2 2
+    constexpr uint32_t DivideAndRoundUp(uint32_t value, uint32_t alignment)
+    {
+        AZ_Assert(alignment != 0, "0 is an invalid multiple to round to.");
+        AZ_Assert(
+            std::numeric_limits<uint32_t>::max() - value >= alignment,
+            "value '%" PRIu32 "' and alignment '%" PRIu32 "' will overflow when added together during DivideAndRoundUp.");
+        return (value + alignment - 1) / alignment;
+    }
+
+    constexpr uint64_t DivideAndRoundUp(uint64_t value, uint64_t alignment)
+    {
+        AZ_Assert(alignment != 0, "0 is an invalid multiple to round to.");
+        AZ_Assert(
+            std::numeric_limits<uint64_t>::max() - value >= alignment,
+            "value '%" PRIu64 "' and alignment '%" PRIu64 "' will overflow when added together during DivideAndRoundUp.");
+        return (value + alignment - 1) / alignment;
+    }
```

Comment:
These functions are in the same vein as SizeAlignUp/SizeAlignDown and PointerAlignUp/PointerAlignDown in [base.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/base.h#L180-L190)